### DISCUSSION TITLE
Where to use ipxe.efidrv?

### DISCUSSION THREAD

**Discussion by ccy** [ipxe/ipxe#724] [General]
I notice the ipxe make file has these entries:

```
bin-i386-efi/ipxe.efi bin-i386-efi/ipxe.efidrv \
bin-x86_64-efi/ipxe.efi bin-x86_64-efi/ipxe.efidrv \
```
Just wondering how to use `ipxe.efidrv`?

**Comment by NiKiZe** [comment]
The same way as you did with RtkUndi ;)

**Reply by ccy** [reply]
I build `ipxe.efidrv` and try to load from UEFI built-in shell:
```
FS0:\> load ipxe.efidrv
Image 'FS0:\ipxe-shell.efidrv' loaded at B620D000 - Success

FS0:\> drivers
...
20C 018C2334 B - - 10 10 ipxe.efidrv           \ipxe.efidrv
```
The `ipxe.efidrv` loaded.

But what can I do next?

**Reply by stappersg** [reply]
`load ipxe.efidrv`, `load RtkUndiDxe.efi` and `drivers` to show / to proof
> same way as with RtkUndi

**Reply by ccy** [reply]
I execute `load ipxe.efidrv`, `load RtkUndiDxe.efi`.  I can see both drivers show in `drivers`.

So what is the purpose of `ipxe.efidrv`?  I don't understand what can I do with `ipxe.efidrv`?

**Reply by NiKiZe** [reply]
It gives you support for NICs that has native drivers in iPXE in your efi environment. You need a network stack in your efi environment to use it.

So instead of developing pure efi driver for a NIC, a vendor could implement it in iPXE and get both efi and pcbios UNDI stack rom from one codebase.

**Reply by ccy** [reply]
Is that possible to embed the `RtkUndiDxe.efi` into `ipxe.efi`?  So only a single file is needed for deployment.
